### DISCUSSION TITLE
Source codes...

### DISCUSSION THREAD

**Discussion by tnagy294** [Jyers/Marlin#555] [General Discussion]
Just a note...would it be possible to change the extruder rotation direction with some user settings throuch the display or it has to be done by compiling the firmware? If only by firmware could you (the author) share the config (source) files set up for FW 4.2.2 board, BLTouch 5x5 (not HS), I would like to know (and sure not just me) what are all the needed settings to compile a FW like E3V2-BLTouch-5x5-v4.2.2.bin. The source code you use to compile the final bin file would help very much, it's not clear to me what are all the necessary setings that must be set. Many Thanks. Tibor

**Comment by Jyers** [comment]
I don't believe this can be moved to an on screen option. The configuration files for what you are looking for can be found in "Configuration Files\E3V2 Templates\BLTouch-3x3" just change `GRID_MAX_POINTS_X` from 3 to 5

**Reply by tnagy294** [reply]
I'll take look, thank you...

**Reply by tnagy294** [reply]
Well, I looked at it . changed points to 5, first run got error...needed to set advanced-pause-resumed...and after taht I get error:
Compiling .pio\build\STM32F103RET6_creality\lib3df\EEPROM\EEPROM.cpp.o
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp: In member function 'void CrealityDWINClass::Draw_Print_ProgressRemain()':
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp:669:31: error: 'class MarlinUI' has no member named 'get_remaining_time'
   uint16_t remainingtime = ui.get_remaining_time();
                               ^~~~~~~~~~~~~~~~~~
Archiving .pio\build\STM32F103RET6_creality\lib4d8\libSTM32ADC.a
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp: In member function 'void CrealityDWINClass::Start_Print(bool)':
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp:4678:8: error: 'class MarlinUI' has no member named 'set_progress'; did you mean '_get_progress'?
     ui.set_progress(0);
        ^~~~~~~~~~~~
        _get_progress
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp:4679:8: error: 'class MarlinUI' has no member named 'set_remaining_time'
     ui.set_remaining_time(0);
        ^~~~~~~~~~~~~~~~~~
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp: In member function 'void CrealityDWINClass::Stop_Print()':
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp:4688:6: error: 'class MarlinUI' has no member named 'set_progress'; did you mean '_get_progress'?
   ui.set_progress(100 * (PROGRESS_SCALE));
      ^~~~~~~~~~~~
      _get_progress
Marlin\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp:4689:6: error: 'class MarlinUI' has no member named 'set_remaining_time'
   ui.set_remaining_time(0);
      ^~~~~~~~~~~~~~~~~~
*** [.pio\build\STM32F103RET6_creality\src\src\lcd\extui\lib\creality_dwin\creality_dwin.cpp.o] Error 1
================================================================================= [FAILED] Took 7.17 seconds =================================================================================
Environment             Status    Duration
----------------------  --------  ------------
STM32F103RET6_creality  FAILED    00:00:07.173
=========================================================================== 1 failed, 0 succeeded in 00:00:07.173 =========================================================================== 
The terminal process "C:\Users\tnagy\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.

Haven't change anything more, just enabled advance_pause_resume ...don't know why I get this one...

**Reply by tnagy294** [reply]
it seems to me, creality_dwin.cpp is somehow some old version because I cannot find some later lines of code in that, e.g. searching for ADVANCED_TIME_FORMAT I cannot find it in the code (according to ading menu for showing mesh values...by kjeldahl)...so maybe you use some newer file (dwin.cpp) and that's why I get that compile error...but I'm not sure

**Reply by Jyers** [reply]
Are you sure you are grabbing the extensible ui branch of the firmware? The code has moved to a new location but the rewrite branch is still on the github, so that would cause some inconsistency.
